src/middlewares/createInsightsMiddleware.ts

```diff
@@ -126,7 +127,14 @@ export const createInsightsMiddleware: CreateInsightsMiddleware = props => {
           if (onEvent) {
             onEvent(event, _insightsClient);
           } else if (event.insightsMethod) {
-            insightsClient(event.insightsMethod, event.payload);
+            if (hasUserToken()) {
+              insightsClient(event.insightsMethod, event.payload);
+            } else {
+              warning(
+                false,
+                'Cannot send event to Algolia Insights because `userToken` is not set.'
```

Comment:
Can we add an actionable doc link that teaches users how to solve this?

Reply:
https://www.algolia.com/doc/guides/building-search-ui/going-further/send-insights-events/js/#setting-the-usertoken

This might be the doc, but it's for js and this code will be used in Vue InstantSearch, too. What do you think? It's okay to just add the js version of the doc?

Reply:
https://github.com/algolia/instantsearch.js/pull/4724/commits/b0245fa51ca7434ac2093464061bf66401e2931d